momepy/preprocessing.py

```diff
@@ -740,3 +746,327 @@ def _get_extrapolated_line(coords, tolerance, point=False):
     if point:
         return b
     return pygeos.linestrings([a, b])
+
+
+def _polygonize_ifnone(edges, polys):
+    if polys is None:
+        pre_polys = polygonize(edges.geometry)
+        polys = gpd.GeoDataFrame(geometry=[g for g in pre_polys], crs=edges.crs)
+    return polys
+
+
+def _selecting_rabs_from_poly(
+    gdf, circom_threshold=0.7, area_threshold=0.85, include_adjacent=True
+):
+    """
+    From a GeoDataFrame of polygons, returns a GDF of polygons that are
+    above the Circular Compaactness threshold.
+
+    Return
+    ________
+    GeoDataFrames : round abouts and adjacent polygons
+    """
+    # calculate parameters
+    gdf.loc[:, "area"] = gdf.geometry.area
+    circom_serie = CircularCompactness(gdf, "area").series
+
+    # selecting round about polygons based on compactness
+    rab = gdf[circom_serie > circom_threshold]
+    # exclude those above the area threshold
+    area_threshold_val = gdf.area.quantile(area_threshold)
+    rab = rab[rab["area"] < area_threshold_val]
```

Comment:
Line 770 assigns a new column to gdf, which may be unwanted if the user passed it themselves. Try this (untested):

```suggestion
    area = gdf.area
    circom = CircularCompactness(gdf, area).series
    
    # selecting round about polygons based on compactness
    mask = circom_serie > circom_threshold
    rab  = gdf[mask]
    # exclude those above the area threshold
    area_threshold_val = area.quantile(area_threshold)
    rab = rab[area[mask] < area_threshold_val]
```

Reply:
I did try that earlier. 
The reason it's useful to have it as a column is that a bit further (line 792) I use the column to define adjacent_polys.
But I will create an `if` clause for the user to provide the column name in case it already exists.

Reply:
The line 792 is not using this column. Since you are calling it as `gdf.area` and not `gdf['area']`, the geopandas attribute measuring the area has a precedence.

Reply:
ups ! just saw your (quick) reply.
I submitted a quick enhancement to accommodate for `area_col` as input. 

What was happening on line 792 (now 803) is that I was comparing the areas of the possible `rabs` against the areas of the possible `adjacent_polys` after the spatial join.

let me know it this too cumbersome and I can revert to something different